src/receive_imf.rs

```diff
@@ -827,6 +827,11 @@ pub(crate) async fn receive_imf_inner(
     if let Some(ref sync_items) = mime_parser.sync_items {
         if from_id == ContactId::SELF {
             if mime_parser.was_encrypted() {
+                // Receiving encrypted message from self updates primary transport.
+                context
+                    .sql
+                    .set_raw_config("configured_addr", Some(&mime_parser.from.addr))
+                    .await?;
```

Comment:
do I understand it correctly that this handles sync of changing the primary transport? like only primary transport sends sync messages, so if we receive sync messages from a different address then we can assume that this address is the new primary transport?